libsolidity/formal/SMTEncoder.cpp

```diff
@@ -883,6 +886,102 @@ void SMTEncoder::endVisit(ElementaryTypeNameExpression const& _typeName)
 	m_context.createExpression(_typeName, result.second);
 }
 
+namespace
+{
+unsigned dimensionCount(TypePointer _type)
+{
+	unsigned dimensionCount = 0;
+	while (_type->category() == Type::Category::Array || _type->category() == Type::Category::Mapping)
+	{
+		++dimensionCount;
+		if (_type->category() == Type::Category::Array)
+			_type = dynamic_cast<ArrayType const *>(_type)->baseType();
+		else if (_type->category() == Type::Category::Mapping)
+			_type = dynamic_cast<MappingType const *>(_type)->valueType();
+		else
+			solAssert(false, "");
+	}
+	return dimensionCount;
+}
+
+vector<string> structGetterReturnedMembers(StructType const* _structType)
+{
+	vector<string> returnedMembers;
+	for (auto const& member: _structType->nativeMembers(nullptr))
+	{
+		// TODO: Figure out which types are returned by the getter and which are ignored
+		if (auto category = member.type->category(); category == Type::Category::Array || category == Type::Category::Mapping)
+			continue;
+		returnedMembers.push_back(member.name);
+	}
+	return returnedMembers;
+}
+
+}
+
+void SMTEncoder::visitPublicGetter(FunctionCall const& _funCall)
+{
+	MemberAccess const& access = dynamic_cast<MemberAccess const&>(_funCall.expression());
+	auto var = dynamic_cast<VariableDeclaration const*>(access.annotation().referencedDeclaration);
+	solAssert(var, "");
+	solAssert(m_context.knownExpression(_funCall), "");
+	auto actualArguments = applyMap(_funCall.arguments(), [this](auto const& argExpr) { return expr(*argExpr); });
+	TypePointer type = var->type();
+	auto handleValueType = [&]() {
+		solAssert(actualArguments.empty(), "");
+		m_context.addAssertion(currentValue(*var) == expr(_funCall));
+	};
+	switch (type->category())
+	{
+		case Type::Category::Bool:
+		case Type::Category::Integer:
+		case Type::Category::Address:
+		case Type::Category::FixedBytes:
+		case Type::Category::Enum:
+		case Type::Category::FixedPoint:
+		case Type::Category::Function:
+		case Type::Category::Contract:
+			handleValueType();
+			break;
+		case Type::Category::Array:
+		case Type::Category::Mapping:
+		{
+			if (auto arrayType = dynamic_cast<ArrayType const*>(type); arrayType && arrayType->isByteArray())
+			{
+				handleValueType();
+				break;
+			}
+			solAssert(dimensionCount(type) == actualArguments.size(), "");
+			smtutil::Expression exprVal = currentValue(*var);
+			for (auto const& arg: actualArguments)
+			{
+				exprVal = smtutil::Expression::select(
+					smtutil::Expression::tuple_get(exprVal, 0),
+					arg
+				);
+			}
+			m_context.addAssertion(expr(_funCall) == exprVal);
```

Comment:
```suggestion
			defineExpr(_funCall, exprVal);
```
Just to make sure the SSA index is increased

Reply:
Changed here and also for the value types. Not sure about the case for the `struct` though.